fetch/api/request/destination/fetch-destination.https.html

```diff
@@ -194,6 +194,44 @@
   });
 }, 'HTMLLinkElement with rel=stylesheet fetches with a "style" Request.destination');
 
+// Import declaration with `type: "css"` - style destination
+promise_test(async t => {
+  await new Promise((resolve, reject) => {
+      frame.contentWindow.onerror = reject;
+
+      let node = frame.contentWindow.document.createElement("script");
+      node.onload = resolve;
+      node.onerror = reject;
+      node.src = "import-declaration-type-css.js";
+      node.type = "module";
+      frame.contentWindow.document.body.appendChild(node);
+  }).then(() => {
+    frame.contentWindow.onerror = null;
+  }, (e) => {
+      assert_unreached("Fetch errored.", e);
+  });
+}, 'Import declaration with `type: "css"` fetches with a "style" Request.destination');
+
+// JSON destination
+///////////////////
+
+// Import declaration with `type: "json"` - json destination
+promise_test(async t => {
+  await new Promise((resolve, reject) => {
+      frame.contentWindow.onerror = reject;
+      let node = frame.contentWindow.document.createElement("script");
+      node.onload = resolve;
+      node.onerror = reject;
+      node.src = "import-declaration-type-json.js";
+      node.type = "module";
+      frame.contentWindow.document.body.appendChild(node);
+  }).then(() => {
+    frame.contentWindow.onerror = null;
+  }, (e) => {
```

Comment:
in theory, there should be no need to have a failureCallback here. Just returning the new promise at line 220 should do the trick.